native/cocos/platform/java/modules/SystemWindow.cpp

```diff
@@ -57,11 +57,15 @@ void SystemWindow::setWindowHandle(void *handle) {
 #if (CC_PLATFORM == CC_PLATFORM_ANDROID)
     //The getWindowHandle interface may have been called earlier, causing _handleMutex to be occupied all the time.
     bool lockSuccess = _handleMutex.try_lock();
-    bool needNotify = _windowHandle == nullptr;
     _windowHandle = handle;
-    if (needNotify) {
+    if (!lockSuccess) {
         _windowHandlePromise.set_value();
     }
+    if (_windowHandle) {
+        auto *nativeWindow = static_cast<ANativeWindow *>(_windowHandle);
+        _width = ANativeWindow_getWidth(nativeWindow);
+        _height = ANativeWindow_getHeight(nativeWindow);
```

Comment:
setWindowHandle should not do other things.

Reply:
updated